tests/main.test.js

```diff
@@ -56,5 +57,54 @@ describe("Verify Getters", () => {
             expectKeys(json_data, return_keys);
             expect(Object.keys(json_data["data"]).length).toBe(max_datapoints);
         });
+
+        test(`/${endpoint} Empty page load`, async() => {
+            const all_data = await cosi_request()
+                                    .get(`/${endpoint}`)
+                                    .expect(200)
+                                    .query({page: Number.MAX_SAFE_INTEGER})
+                                    .expect("Content-Type", /json/);
+            let json_data = JSON.parse(all_data.text);
+            expect(Object.keys(json_data["data"]).length).toBe(0);
+        });
+
+        test(`/${endpoint} Invalid page load`, async() => {
+            const fake_pages = ["cosi", "-1", "!@#$%^&*()-_+=`"];
+            for (const page of fake_pages){
+                const all_data = await cosi_request()
+                                    .get(`/${endpoint}`)
+                                    .query({page: `${page}`})
+                                    .expect(200)
+                                    .expect("Content-Type", /json/);
+                let json_data = JSON.parse(all_data.text);
+                expect(json_data["page"]).toBe(0);
+                expect(Object.keys(json_data["data"]).length).toBe(max_datapoints);
+                expect(json_data["total_pages"]).toBe(Math.ceil(total_datapoints_per_table/max_datapoints));
+            }
+        });
+
+        test(`/${endpoint} Correct page count`, async() => {
```

Comment:
Do we also want to do a quick sanity check and make sure that two consecutive pages do not have duplicate data? (as is the case for generators unless there was a hiccup with the overlapping of pages.)

Reply:
Good point! Implemented.